### DISCUSSION TITLE
Deploy on GitHub

### DISCUSSION THREAD

**Discussion by StefanKarneback** [community/community#59821] [New to GitHub] [answered]
Hi
I have a web page that works on localhost. I can push the local repository to GitHub. When trying to save (and thus deploy the page) it gives error 404 (page ot found). When I had Readme.md the page showed only text from Readme and that was it. When I deleted Readme the page is blank.

I have two branches, main and gh-pages. README is deleted from main which gives error 404.
README exists in gh-pages and that text is presented when deploying.

How can I find a solution? Where shall I search for faulty files?

**Comment by StefanKarneback** [comment, marked as answer]
Hi again
I found a command that made my page become published (deployed): npm run deploy -- -m "Deploy React app to GitHub Pages".
So now it's working, I hope even the next time.

**Reply by GhostOf0days** [reply]
Just as advice, use Netlify (for more simple projects), Heroku, or Vercel. Avoid Github Pages.

**Reply by StefanKarneback** [reply]
I will have a look at them. Thanks.

**Reply by dianegeorge** [reply]
Idk

**Comment by Uliana200407** [comment]
Also that maybe an issue with deploying , it sound mad, but you can try to clear your browser - it helped me

**Reply by StefanKarneback** [reply]
Do you mean just clear the history?

**Comment by Uliana200407** [comment]
Yep,just clean the history in browser

пт, 13 жовт. 2023 р. о 13:29 StefanKarneback ***@***.***>
пише:
